OPHD/Things/Structures/FoodProduction.h

```diff
@@ -22,18 +28,34 @@ class FoodProduction : public Structure
 		stringTable[{1, 0}].text = std::to_string(mFoodLevel) + " / " + std::to_string(foodCapacity());
 
 		stringTable[{0, 1}].text = "Production Rate:";
-		stringTable[{1, 1}].text = std::to_string(calculateProduction());
+		stringTable[{1, 1}].text = std::to_string(mFoodProductionRate);
 
 		return stringTable;
 	}
 
 	int foodLevel() const { return mFoodLevel; }
 	void foodLevel(int level) { mFoodLevel = std::clamp(level, 0, foodCapacity()); }
 
-	virtual int foodCapacity() = 0;
+	int foodCapacity() const { return mFoodCapacity; }
 
-protected:
-	virtual int calculateProduction() = 0;
+	void produce()
+	{
+		if (structure().operational())
+		{
+			foodLevel(foodLevel() + mFoodProductionRate);
+			if (mFoodProductionRate && mFoodLevel == mFoodCapacity)
+			{
+				structure().idle(IdleReason::InternalStorageFull);
+			}
+		}
+		else if (structure().disabled())
+		{
+			foodLevel(0);
```

Comment:
this was done in `Agridome::disabledStateSet` before. Now being cleared in the food-growing phase of the new turn.